Update spring boot

lleifermann opened the pull request:
> Closes #6 

I updated spring boot to 2.0.1 and fixed an array of now broken things. This includes:
- findOne is no longer usable. Instead you now have to use findOneById which now returns an optional. Please especially look over those occurences where i tried to fix this inside streams. 
- WebMvcConfigurerAdapter is now deprecated an can be replaced by an implementation of WebMvcConfigurer.
- This still needs to be changed but i wanted feedback first. The annotation NotBlank is  
deprecated too. Any substitutes for this?
-Since the Hikari Datasource is now standard, do we need the current implementation of it?

EDIT: 

- As @DennisVonDerBey pointed out. Its possible to substitute Notblank from javax.validation.constraints. I did that accordingly. 
- The AbstractWebSocketMessageBrokerConfigurer was also deprecated and could be swapped with an implementation of WebSocketMessageBrokerConfigurer
- I removed all traces of our direct hikari implementation, since this is now directly supported by spring boot. I just deleted all our usages and it seems to work fine.

###### References

* [Spring-Boot 2.0 Migration Guide][sb2-mig-guide]
* [Spring Boot Security 2.0 Migration Guide][sb2-sec-mig-guide]

[sb2-mig-guide]: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide
[sb2-sec-mig-guide]: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Security-2.0

lleifermann committed:
Update imports

lleifermann committed:
Update to new repository methods and return types

lleifermann committed:
Update dependencies

lleifermann committed:
Adapt to new and deprecated properties

fermentfan commented on src/main/java/de/communicode/communikey/service/KeyService.java:
```diff
@@ -32,15 +32,12 @@
-import de.communicode.communikey.exception.HashidNotValidException;
-import de.communicode.communikey.exception.KeyNotAccessibleByUserException;
-import de.communicode.communikey.exception.UserEncryptedPasswordNotFoundException;
+import de.communicode.communikey.exception.*;
```
No asterisk import

fermentfan commented on src/main/java/de/communicode/communikey/service/UserService.java:
```diff
@@ -437,13 +439,16 @@ public User updateAuthorities(String login, Set<String> payload) {
             .map(user -> {
                 Set<Authority> payloadAuthorities = payload.stream()
-                    .map(authorityRepository::findOne)
+                    .map(id -> authorityRepository.findById(id).orElseThrow(() -> new AuthorityNotFoundException(id)))
```
Is this really necessary? Shouldn't it suffice to change the method?

fermentfan commented on src/main/java/de/communicode/communikey/service/UserService.java:
```diff
@@ -437,13 +439,16 @@ public User updateAuthorities(String login, Set<String> payload) {
                     payload.stream()
-                        .map(authorityRepository::findOne)
-                        .forEach(user::addAuthority);
+                        .map(authorityRepository::findById)
```
Duplicate code, already on it

fermentfan commented on src/main/java/de/communicode/communikey/security/SecurityAuditorAware.java:
```diff
@@ -43,11 +45,11 @@ public SecurityAuditorAware(CommunikeyProperties communikeyProperties) {
+            return ofNullable(userName);
         }
-        return communikeyProperties.getSecurity().getRoot().getLogin();
+        return ofNullable(communikeyProperties.getSecurity().getRoot().getLogin());
```
```return Optional.empty();```

lleifermann committed:
Removed explicit implementations of HikariCP

svengreb commented:
Changed the target branch to the new default branch `develop`.

svengreb committed:
Reimplement the data source configuration

The class is necessary to initialize the OAuth2 database schema and
to enable the JPA repository auditing. According to the Spring Boot
documentation Hibernate JPA managment only automatically runs schema
initialization scripts found in resources path when Hibernate JPA DDL
management is not enabled (1).
The data source configuration itself is done via autoconfiguration to
let Spring autodetect the DB type, optimal dialect and other settings.

This commit also removes the now unused HikariCP properties from the
"application*.yml" configuration.
The recommended properties for a better MySQL performance have also
been restored (2).

References:

  (1) https://docs.spring.io/spring-boot/docs/current/reference/html/howto-database-initialization.html#howto-initialize-a-database-using-spring-jdbc
  (2) https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration

GH-6

svengreb committed:
Update ErrorAttributes bean

The Spring class "ServerRequest" has been moved from the
"spring-boot-web" to the "spring-boot-webflux" package. Also removed a
unnecessary check if the attribute map contains the "message" key.

GH-6

svengreb commented on src/main/java/de/communicode/communikey/config/DataSourceConfig.java:
```diff
@@ -17,91 +17,69 @@
-@EnableJpaAuditing(auditorAwareRef = "securityAuditorAware")
-@EnableTransactionManagement
+@EnableJpaAuditing(auditorAwareRef = "securityAuditorAware", dateTimeProviderRef="dateTimeProvider")
 public class DataSourceConfig {
```
Restored and updated in f5a271c6da073e532d72db82c85e2c6e9c8d1968.

svengreb commented on src/main/java/de/communicode/communikey/exception/GlobalControllerExceptionHandler.java:
```diff
@@ -163,13 +163,11 @@
     public ErrorAttributes errorAttributes() {
```
Updated in 7e6ad6831b5a177f0d1731bf0e97f6448e88bb57.

svengreb committed:
Use "Instant" class instead of "ZonedDateTime" for auditing

As of Spring Boot 2.0 the "java.time.ZonedDateTime" class is not
supported for JPA auditing (AuditAware) out-of-the-box and required to
register a DateTimeProvider bean. See reported issues DATAJPA-1242 (1)
and GH-10743 (2).

The usage of "ZonedDateTime" for auditing has been advised against
because the time zone is really more of a presentation layer concern.
The use of "LocalDateTime" is problematic as well. It does not represent
a point of time but does so only together with a timezone. The current
implementation of Spring "CurrentDateTimeProvider" uses the JVM default
timezone. If an application runs in a cluster across time zones this
results in values of "LocalDateTime" values that can't be compared anymore.

The recommended class to use is "java.time.Instant" instead because there
is no timezone involved, yet it specifies a point in time. For details
see issue DATACMNS-1243 (3) which added converters for
"Instant" <-> "LocalDateTime".

References:

  (1) https://jira.spring.io/browse/DATAJPA-1242
  (2) https://github.com/spring-projects/spring-boot/issues/10743
  (3) https://jira.spring.io/browse/DATACMNS-1243

GH-6

svengreb committed:
Remove no longer needed Jackson Hibernate 5 module

The "jackson-datatype-hibernate5" module has been used in earlier
versions to handle eagerly fetched entities when the API used to
recursively serialize objects in collections instead of IDs. Is was
used for the "@JsonManagedReference" and "@JsonBackReference"
annotations (1) added to collections within the entity classes to handle
bidirectional relationships. This was a bad API design (recursive tree
serialization for nested key categories) and has already been changed.
This commit removes the still loaded Jackson module and cleanes up the
Maven dependency.

References:

  (1) http://www.baeldung.com/jackson-bidirectional-relationships-and-infinite-recursion

GH-6

svengreb committed:
Replace single "jackson-modules-java8" with "spring-boot-starter-json"

Replaced the single Jackson Java 8 modules (1) with the new
"spring-boot-starter-json" package added in Spring Boot 2.0 (2). It provides,
next to "jackson-databind", the modules

* jackson-datatype-jdk8
* jackson-datatype-jsr310
* jackson-module-parameter-names

This commit also removes a duplicate Jackson property which is already
configured in the "JacksonConfig" class.

References:

  (1) https://github.com/FasterXML/jackson-modules-java8
  (2) https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide#jackson--json-support

GH-6

svengreb committed:
Update various dependency versions

Bumped all outdated dependencies to the latest version. This
includes Maven plugins, Spring Boot packages and all other production
and test libraries.
This commit also replaces project version constant with a hardcoded
value and fixes some comment indentions.

GH-6

svengreb committed:
Remove unused SpringFox Swagger dependency

The automatic REST API docs generation has never been used and wil be
available on the project documentation website.

GH-6

svengreb committed:
Add "@Validated" annotation to custom property class

According to the Spring Boot 2.0 migration guide (1) it is now mandatory
that the "@ConfigurationProperties" object is annotated with
"@Validated" to turn on validation.

References:

  (1) https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide#configurationproperties-validation

GH-6

svengreb reviewed (approved):
A detailed documentation will be available in the next version release changelog. Also of course the changes are documented in the commit messages.
We are aware that the CI builds are failing which will be resolved in #2 and #8.
